tests/test_indexing.py

```diff
@@ -647,3 +650,123 @@ def test_random_with_replace(self):
         pairs2 = index_cl2.index(self.a)
         assert len(pairs2) == 1000
         assert not pairs2.is_unique
+
+
+# NeighbourhoodBlock
+
+class TestNeighbouroodBlock(TestData):
+    """General unittest for the NeighbourhoodBlocking indexing class."""
+
+    @classmethod
+    def setup_class(cls):
+
+        TestData.setup_class()
+
+        def incomplete_df_copy(df, nan_proportion=0.1):
+            'copy of DataFrame with some cells set to NaN'
+            nan_count = int(round(len(df) * nan_proportion))
+            def with_nulls(vals):
+                vals = vals.copy()
+                vals.iloc[np.random.choice(len(df), size=nan_count, replace=False)] = np.nan
+                return vals
+            return df.copy() if nan_count <= 0 else df.apply(with_nulls)
+
+        np.random.seed(0)
+        cls.incomplete_a = incomplete_df_copy(cls.a)
+        cls.incomplete_b = incomplete_df_copy(cls.b)
+
+    def assert_index_comparisons(self, pairwise_comparison, indexers, *args, **kwargs):
```

Comment:
Really nice. I think this can be useful for other tests as well. Thanks!

Reply:
Thanks